test/integration/admin_analytics_test.rb

```diff
@@ -0,0 +1,67 @@
+require 'test_helper'
+require 'capybara/rails'
+
+class AdminAnalyticsTest < ActionDispatch::IntegrationTest
+  include Capybara::DSL
+
+  def log_out
+    GDS::SSO.test_user = nil
+  end
+
+  setup do
+    ENV['GDS_SSO_MOCK_INVALID'] = '1'
+  end
+
+  teardown do
+    ENV.delete('GDS_SSO_MOCK_INVALID')
+  end
+
+  # GA is noisy in tests because all GA calls become console.log so we
+  # don't want it enabled for all tests, use this to selectively turn it on
+  def with_ga_enabled
+    begin
+      GovukAdminTemplate.configure { |c| c.enable_google_analytics_in_tests = true }
+      yield
+    ensure
+      GovukAdminTemplate.configure { |c| c.enable_google_analytics_in_tests = false }
+    end
+  end
+
+  def refute_dimension_is_set(dimension)
+    refute_match(/#{Regexp.escape("GOVUKAdmin.setDimension(#{dimension}")}/, page.body)
+  end
+
+  def assert_dimension_is_set(dimension, with_value: nil)
+    dimension_set_js_code = "GOVUKAdmin.setDimension(#{dimension}"
+    dimension_set_js_code += ", \"#{with_value}\")" if with_value.present?
```

Comment:
So if there's no `with_value` present, we're missing a closing bracket for the previous line ?

Would a better solution be:
`dimension_set_js_code +=  with_value.present? ? ", \"#{with_value}\")" : ")"`
?

It hasn't failed because we do regex matching.